zone/zonedb.cpp

```diff
@@ -943,38 +944,46 @@ bool ZoneDatabase::LoadCharacterMaterialColor(uint32 character_id, PlayerProfile
 
 bool ZoneDatabase::LoadCharacterBandolier(uint32 character_id, PlayerProfile_Struct* pp)
 {
-	std::string query = StringFormat("SELECT `bandolier_id`, `bandolier_slot`, `item_id`, `icon`, `bandolier_name` FROM `character_bandolier` WHERE `id` = %u LIMIT %u",
-		character_id, EQ::profile::BANDOLIERS_SIZE);
-	auto results = database.QueryDatabase(query); int i = 0; int r = 0; int si = 0;
-	for (i = 0; i < EQ::profile::BANDOLIERS_SIZE; i++) {
+	const auto& l = CharacterBandolierRepository::GetWhere(
+		database,
+		fmt::format(
+			"`id` = {} LIMIT {}",
+			character_id,
+			EQ::profile::BANDOLIERS_SIZE
+		)
+	);
+
+	for (int i = 0; i < EQ::profile::BANDOLIERS_SIZE; i++) {
 		pp->bandoliers[i].Name[0] = '\0';
+
 		for (int si = 0; si < EQ::profile::BANDOLIER_ITEM_COUNT; si++) {
-			pp->bandoliers[i].Items[si].ID = 0;
+			pp->bandoliers[i].Items[si].ID   = 0;
 			pp->bandoliers[i].Items[si].Icon = 0;
+
 			pp->bandoliers[i].Items[si].Name[0] = '\0';
 		}
 	}
 
-	for (auto& row = results.begin(); row != results.end(); ++row) {
-		r = 0;
-		i = Strings::ToInt(row[r]); /* Bandolier ID */ r++;
-		si = Strings::ToInt(row[r]); /* Bandolier Slot */ r++;
-
-		const EQ::ItemData* item_data = database.GetItem(Strings::ToInt(row[r]));
+	for (const auto& e : l) {
+		const auto* item_data = database.GetItem(e.item_id);
 		if (item_data) {
-			pp->bandoliers[i].Items[si].ID = item_data->ID; r++;
-			pp->bandoliers[i].Items[si].Icon = Strings::ToInt(row[r]); r++; // Must use db value in case an Ornamentation is assigned
-			strncpy(pp->bandoliers[i].Items[si].Name, item_data->Name, 64);
-		}
-		else {
-			pp->bandoliers[i].Items[si].ID = 0; r++;
-			pp->bandoliers[i].Items[si].Icon = 0; r++;
-			pp->bandoliers[i].Items[si].Name[0] = '\0';
-		}
-		strcpy(pp->bandoliers[i].Name, row[r]);  r++;
+			pp->bandoliers[e.bandolier_id].Items[e.bandolier_slot].ID   = item_data->ID;
+			pp->bandoliers[e.bandolier_id].Items[e.bandolier_slot].Icon = e.icon;
+
+			strn0cpy(
+				pp->bandoliers[e.bandolier_id].Items[e.bandolier_slot].Name,
+				item_data->Name,
+				sizeof(pp->bandoliers[e.bandolier_id].Items[e.bandolier_slot].Name)
```

Comment:
Same thing, we don't want to be using `sizeof` of something that isn't even set yet. Just use 64 or `item_data->Name`